cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -615,12 +620,40 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
         return declStmt
     }
 
+    private fun addExternallyImportedToAnalysis(importName: Name) {
+        ctx?.let { ctx ->
+            var currentName = importName
+            while (currentName.isNotEmpty()) {
+                var importPath = currentName.toString().replace(language.namespaceDelimiter, "/")
+
+                language.fileExtensions.forEach { fileExtension ->
+
+                    // Includes a file in the analysis, if it has the root path and
+                    ctx.externalSources
+                        .firstOrNull {
+                            val relPath =
+                                it.relativeTo(
+                                        Util.getRootPath(it, ctx.config.includePaths).toFile()
+                                    )
+                                    .path
+                            val ending = "." + fileExtension
+                            relPath == importPath + ending ||
+                                relPath ==
+                                    importPath + "/" + PythonLanguage.IDENTIFIER_INIT + ending
+                        }
+                        ?.let { ctx.importedSources += it }
+                }
+                currentName = currentName.parent ?: Name("")
```

Comment:
creating a new empty name is uncecessary, you can just use null instead and do a nil check in the while loop.

Reply:
Removed that Name creation and changed the loop condition.